ui/selectors/selectors.js

```diff
@@ -1170,9 +1170,15 @@ export function getNetworkConfigurations(state) {
 
 export function getCurrentNetwork(state) {
   const allNetworks = getAllNetworks(state);
+  const providerConfig = getProviderConfig(state);
   const currentChainId = getCurrentChainId(state);
 
-  return allNetworks.find((network) => network.chainId === currentChainId);
+  const filter =
+    providerConfig.type === 'rpc'
+      ? (network) => network.id === providerConfig.id
+      : (network) => network.chainId === currentChainId;
```

Comment:
Thoughts on using `provider.type` instead? `chainId` works, but it makes the result order-dependent. `type` is reliably unique for built-in networks, so that should make this a little more reliable even if the order changes later.

It looks like `providerConfig.type` is available on the network list entries as `network.providerType`

```suggestion
      : (network) => network.providerType === providerConfig.type;
```

Alternatively, since you've set the `id` field to the type for all of the built-in networks, maybe this would work:

```suggestion
      : (network) => network.id === providerConfig.type;
```

Reply:
Awesome idea!

Reply:
Using `id` might be more forwards-compatible if we start introducing more custom RPC fields alongside rpc URL, making it no longer unique. We did get a request to support additional headers, where the header might determine the chain.